backend/app/registry/settings.py

```
@@ -215,6 +220,9 @@ def write(
             # bool is an int subclass — reject it explicitly so `true` can't store as 1.
             if isinstance(value, bool) or not isinstance(value, int) or value < 0:
                 raise ValueError(f"invalid idle_timeout_s: {value!r} (seconds, ≥ 0)")
+        if key == "usage_retention_days":
+            if isinstance(value, bool) or not isinstance(value, int) or value < 0:
```

Ruff (0.16.3)

[warning] Use a single `if` statement instead of nested `if` statements

(SIM102)